allenact/utils/system.py

```diff
@@ -152,3 +152,16 @@ def filter(self, record):
             or ALLENACT_INSTALL_DIR in record.pathname
             or "main" in record.pathname
         )
+
+
+class ImportChecker:
+    def __init__(self, msg=None):
+        self.msg = msg
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, value, traceback):
+        if exc_type == ModuleNotFoundError and self.msg is not None:
+            value.msg += self.msg
+        return exc_type is None
```

Comment:
I'm not super familiar with context managers for exception handling. I assume that we don't have to re-raise the error here right? I.e. we don't have to do something like `raise value`?

Reply:
We don't need to raise anything, because it will be forwarded if we return False. I initially made a different design, but I found this easier to use.